rewrite-test/src/main/kotlin/org/openrewrite/java/dataflow/DataFlowInsanityTest.kt

```diff
@@ -0,0 +1,162 @@
+package org.openrewrite.java.dataflow
+
+import org.junit.jupiter.api.Test
+import org.openrewrite.Cursor
+import org.openrewrite.ExecutionContext
+import org.openrewrite.java.JavaIsoVisitor
+import org.openrewrite.java.tree.Expression
+import org.openrewrite.test.RecipeSpec
+import org.openrewrite.test.RewriteTest
+
+@Suppress("FunctionName")
+interface DataFlowInsanityTest : RewriteTest {
+    override fun defaults(spec: RecipeSpec) {
+        spec.recipe(RewriteTest.toRecipe {
+            object : JavaIsoVisitor<ExecutionContext>() {
+                override fun visitExpression(expression: Expression, p: ExecutionContext): Expression {
+                    dataflow().findSinks(object : LocalFlowSpec<Expression, Expression>() {
+                        override fun isSource(source: Expression, cursor: Cursor) = true
+
+                        override fun isSink(sink: Expression, cursor: Cursor) = true
+                    })
+                    return super.visitExpression(expression, p)
+                }
+            }
+        })
+        spec.expectedCyclesThatMakeChanges(0).cycles(1)
+    }
+
+    @Test
+    fun `random types doing strange things`() = rewriteRun(
+        java(
+            """
+                import java.util.Collection;
+                import java.util.ArrayList;
+                import java.util.function.Supplier;
+
+                @SuppressWarnings("RedundantSuppression")
+                abstract class Test<P extends Collection> implements Supplier<P> {
+                    Object field;
+                    Object ternaryStrangeness = conditional() ? get() : get().stream();
+                    static Boolean conditional() {
+                        return null;
+                    }
+                    static {
+                        Supplier s = new Test<Collection>() {
+                            @Override
+                            public Collection get() {
+                                return new ArrayList<>();
+                            }
+                        };
+                    }
+                    Test() {
+                        Collection c = new ArrayList();
+                        c.add(1);
+                        //noinspection UnusedAssignment
+                        field = c;
+                        this.field = "Over achievements!";
+                    }
+
+                    @Override
+                    public P get() {
+                        return null;
+                    }
+
+                    void test() {
+                        String n = "42";
+                        String o = n;
+                        System.out.println(o);
+                        String p = o;
+                    }
+                }
+            """
+        )
+    )
+
+    @Test
+    fun `goat test`() = rewriteRun(
+        java(
+            """
+package org.openrewrite.java;
```

Comment:
If you don't mind indent these over a couple indents to the right. Margins are trimmed by default. Then merge away!